src/preamble.js

```diff
@@ -1822,13 +1822,11 @@ var runDependencyTracking = {};
 #endif
 
 function getUniqueRunDependency(id) {
-#if ASSERTIONS
   var orig = id;
   while (1) {
     if (!runDependencyTracking[id]) return id;
```

Comment:
Ahhh, now I understand. Yeah, the failing assert also only existed inside `ASSERTIONS`, I went and fixed this in two locations. Reverted this change, adding the `asyncLoad` to call `getUniqueRunDependency()` is enough.

The reason why I found the semantics here odd is that reading a call site of function `getUniqueRunDependency()`, e.g. in https://github.com/kripken/emscripten/blob/master/src/library_fs.js#L1766, makes it read like it should always generate a guaranteed unique name for the dependency, except that it does so only in assertions mode and does nothing in release.  Any case, updated the PR now to restore the assertion guard.